qemu/tests/rng_maxbytes_period.py

```diff
@@ -0,0 +1,85 @@
+import re
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import virt_vm
+from virttest import utils_misc
+
+from aexpect.exceptions import ShellTimeoutError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu virtio-rng device test:
+    1) boot guest with virtio-rng device
+    2) read random data in guest
+    3) check the read data rate
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _is_rngd_running():
+        """
+        Check whether rngd is running
+        """
+        output = session.cmd_output(check_rngd_service)
+        if 'running' not in output:
+            return False
+        return True
+
+    timeout = float(params.get("login_timeout", 360))
+    read_rng_timeout = float(params.get("read_rng_timeout", 3600))
+    read_rng_cmd = params["read_rng_cmd"]
+    max_bytes = params.get("max-bytes_virtio-rng-pci")
+    period = params.get("period_virtio-rng-pci")
+
+    if not max_bytes and not period:
+        test.error("Please specify the expected max-bytes and/or period.")
+    if not max_bytes or not period:
```

Comment:
Not necessary, for line 39 makes sure at least one param is set

Reply:
Line 39: both max_bytes and period are all not defined.
Line 41: only one parameter is defined (either max_bytes or period), negative_test will fall into this part.
Line 54: both max_bytes and period are all defined, positive_test will fall into this part.
Thanks!

Reply:
Got it, so this is part of negative test, and line 68 is another part, and two 'return' for tc termination

Then please ignore my comment right below this one.